controllers/helpdesk.js

```diff
@@ -4,13 +4,292 @@ const router = express.Router();
+				class: 'btn-edit',
+				icon: 'edit',
+				title: res.$t('administration.controller.link.editEntry'),
+			},
+			{
+				link: path + item._id,
+				class: 'btn-disable',
+				icon: 'archive',
+				method: 'delete',
+				title: res.$t('administration.controller.link.completeEntry'),
+			},
+			{
+				link: path + item._id,
+				class: 'btn',
+				icon: 'paper-plane',
+				method: 'post',
+				title: res.$t('administration.controller.link.sendEntryToDevelopmentTeam'),
+			},
+		);
+	}
+	return actions;
+};
+
+/**
+ * creates an event for a created course. following params has to be included in @param data for creating the event:
+ * startDate {Date} - the date the course is first take place
+ * untilDate {Date} -  the date the course is last take place
+ * duration {Number} - the duration of a course lesson
+ * weekday {Number} - from 0 to 6, the weekday the course take place
+ * @param data {object}
+ * @param service {string}
+ * @param req
+ * @param res
+ */
+const createEventsForData = (data, service, req, res) => {
+	// can just run if a calendar service is running on the environment and the course have a teacher
+	if (
+		CALENDAR_SERVICE_ENABLED
+		&& service === 'courses'
+		&& data.teacherIds[0]
+		&& data.times.length > 0
+	) {
+		return Promise.all(
+			data.times.map((time) => api(req).post('/calendar', {
+				json: {
+					summary: data.name,
+					location: res.locals.currentSchoolData.name,
+					description: data.description,
+					startDate: new Date(
+						new Date(data.startDate).getTime() + time.startTime,
+					).toISOString(),
+					duration: time.duration,
+					repeat_until: data.untilDate,
+					frequency: 'WEEKLY',
+					weekday: recurringEventsHelper.getIsoWeekdayForNumber(time.weekday),
+					scopeId: data._id,
+					courseId: data._id,
+					courseTimeId: time._id,
+				},
+				qs: { userId: data.teacherIds[0] },
+			})),
+		);
+	}
+
+	return Promise.resolve(true);
+};
+
+const getUpdateHandler = (service) => function updateHandler(req, res, next) {
+	api(req)
+		.patch(`/${service}/${req.params.id}`, {
+			// TODO: sanitize
+			json: req.body,
+		})
+		.then((data) => {
+			createEventsForData(data, service, req, res).then(() => {
+				res.redirect(cutEditOffUrl(req.header('Referer')));
+			});
+		})
+		.catch((err) => {
+			next(err);
+		});
+};
+
+const getDetailHandler = (service) => function detailHandler(req, res, next) {
+	api(req)
+		.get(`/${service}/${req.params.id}`)
+		.then((data) => {
+			res.json(mapEventProps(data, service));
+		})
+		.catch(next);
+};
+
+/**
+ * Set state to closed of helpdesk problem
+ * @param service usually helpdesk, to disable instead of delete entry
+ * @returns {Function}
+ */
+const getDisableHandler = (service) => function diasableHandler(req, res, next) {
+	api(req)
+		.patch(`/${service}/${req.params.id}`, {
+			json: {
+				state: 'closed',
+				order: 2,
+			},
+		})
+		.then(() => {
+			redirectHelper.safeBackRedirect(req, res);
+		});
+};
+
+
```

Comment:
Codacy found an issue: [More than 1 blank line not allowed.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-client/pullRequest?prid=6046724)